package.json

```diff
@@ -62,12 +62,15 @@
     "jsdom": "^9.12.0",
     "mocha": "^4.0.0",
     "prop-types": "^15.6.0",
-    "react": "^15.6.1",
-    "react-dom": "^15.6.1",
+    "react": "^16.0.0",
+    "react-dom": "^16.0.0",
     "rimraf": "^2.6.2",
     "sinon": "^4.0.0",
     "sinon-sandbox": "^1.0.2"
   },
+  "dependencies": {
+    "semver": "^5.4.1"
```

Comment:
1200 LOC just so we can support both Reacts at the same time? I'd rather semver breaking and backport fixes. Having two different branches is not worth it IMO.

Reply:
Another alternative is letting the default entry point be "render" and providing a new entry point for "hydrate".

Reply:
created a separate function for hydrate

Reply:
Another alternative is calling hydrate when it's present, and otherwise just fall back to render - checking the React version is brittle anyways